filters/model.js

```diff
@@ -0,0 +1,314 @@
+    additionalContent({ renderer, model, content = "" }) {
+      const formattedModelName = FormatHelpers.toPascalCase(model.$id);
+      const properties = model.properties || {};
+      const toStringProperties = Object.keys(properties).map(prop => `"    ${prop}: " + toIndentedString(${FormatHelpers.toCamelCase(prop)}) + "\\n" +`);
+    
+      return content + `
+@Override
+public String toString() {
+  return "class ${formattedModelName} {\\n" +   
+${renderer.indent(renderer.renderBlock(toStringProperties), 4)}
+    "}";
+}
+    
+/**
+ * Convert the given object to string with each line indented by 4 spaces
+ * (except the first line).
+ */
+private String toIndentedString(Object o) {
+  if (o == null) {
+    return "null";
+  }
+  return o.toString().replace("\\n", "\\n    ");
+}`;
+    }
+  }
+}
+
+const EQUAL_AND_HAS_PRESET = {
+  class: {
+    additionalContent({ renderer, model, options, content = '' }) {
+      if (options.disableEqualsHashCode === 'true') return "";
+  
+      const formattedModelName = FormatHelpers.toPascalCase(model.$id);
+      const properties = model.properties || {};
+  
+      const hashProperties = Object.keys(properties).map(prop => FormatHelpers.toCamelCase(prop)).join(', ');
+      const equalProperties = Object.keys(properties).map(prop => {
+        const camelCasedProp = FormatHelpers.toCamelCase(prop);
+        return `Objects.equals(this.${camelCasedProp}, self.${camelCasedProp})`;
+      }).join(' &&\n');
+    
+      return `@Override
+public boolean equals(Object o) {
+  if (this == o) {
+    return true;
+  }
+  if (o == null || getClass() != o.getClass()) {
+    return false;
+  }
+  ${formattedModelName} self = (${formattedModelName}) o;
+    return 
+${renderer.indent(equalProperties, 6)};
+}
+        
+@Override
+public int hashCode() {
+  return Objects.hash(${hashProperties});
+}
+${content}`;
+    }
+  }
+}
+
+const TEMPLATE_PRESET = {
+  class: {
+    property({ renderer, content }) {
+      const annotation = renderer.renderAnnotation('Valid');
+      return renderer.renderBlock([annotation, content]);
+    },
+  },
+}
+
+async function renderJavaModel(schema, schemaName, params, callback) {
+  javaGenerator = new JavaGenerator({ presets: [
+    TEMPLATE_PRESET,
+    INLINE_ONE_ANY_OF,
+    INLINE_ALL_OF,
+    INLINE_ENUM_PRESET,
+
+    JAVA_CONSTRAINTS_PRESET,
+    JAVA_JACKSON_PRESET,
+    JAVA_DESCRIPTION_PRESET,
+
+    CLASS_TO_STRING_PRESET,
+    {
+      preset: EQUAL_AND_HAS_PRESET,
+      options: {
+        disableEqualsHashCode: params.disableEqualsHashCode,
+      }
+    }
+  ] });
+
+  // copy schema
+  const newSchema = Object.assign({}, schema)._json;
+  newSchema.title = newSchema.title || schemaName;
+
+  try {
+    const models = await javaGenerator.generate(newSchema);
+    const result = models[0].result;
+ 
+    callback(null, result);
+  } catch (error) {
+    console.error(error);
+    callback(error);
+  }
+}
+
+filter.renderJavaModel = renderJavaModel;
```

Comment:
I would love some comments here 😅

Reply:
Someone who dont know what the model gen library is will have a hard time understanding this code 😄

Reply:
Yes, I will write, thanks! :)

Reply:
I have the feeling that most of the code here (if not all) should be extracted to its own preset module somewhere. If we want to do the same in the other Java template we have in the future, this code will remain the same and we don't want to duplicate it.

Also, remember, the purpose of the model generation library is to make it easier for template authors, and this is not easy :)

Reply:
@fmvilas Yes I know about common code/presets which we can use in another templates - I created PR with `description`, `jackson`, `constraints` presets. I can also create presets for `hashCode`, `isEqual`, `toString` functions as well. Main problem in this model-generation is that I wanna to have this same way of rendering model as old one, so I had to make this magic with `INLINE_ANY_OF` etc. On the other hand, the presets themselves showed that you can change what you want in the model - of course you must know how it works, which interfaces presets implements etc, but yes, you can do that.

Reply:
I'm curious if @Tenischev would be happy without these. tweaks. Maybe we don't need to reproduce exactly what was there and leave exactly what the model generator is outputting.